### PR TITLE
.github/workflows: separate e2e workflows for 0.34.x and master

### PR DESCRIPTION
**Pull request by tessr**

### CODE REVIEW

File: .github/workflows/e2e-nightly-master.yml [hunk 1]
```diff
@@ -1,5 +1,4 @@
-# Runs randomly generated E2E testnets nightly.
-# Copy of earlier nightly test file, for debugging
+# Runs randomly generated E2E testnets nightly on master
 name: e2e-nightly2
```
Comment (marbar3778) [inline .github/workflows/e2e-nightly-master.yml:2]:
```suggestion
name: e2e-nightly-master
```
? easier to distinguish which one is master.

File: .github/workflows/e2e-nightly-34x.yml [hunk 2]
```diff
@@ -1,7 +1,8 @@
-# Runs randomly generated E2E testnets nightly.
+# Runs randomly generated E2E testnets nightly
+# on the 0.34.x release branch
 name: e2e-nightly
 on:
-  workflow_dispatch: # allow running workflow manually
+  workflow_dispatch: # allow running workflow manually, in theory
```
Comment (erikgrinaker) [inline .github/workflows/e2e-nightly-34x.yml:5]: 😐

File: .github/workflows/e2e-nightly-34x.yml [hunk 3]
```diff
@@ -49,5 +52,21 @@ jobs:
...
+    steps:
+      - name: Notify Slack on success
+        uses: rtCamp/action-slack-notify@ae4223259071871559b6e9d08b24a63d71b3f0c0
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
+          SLACK_CHANNEL: tendermint-internal
+          SLACK_USERNAME: Nightly E2E Tests
+          SLACK_ICON_EMOJI: ':white_check_mark:'
+          SLACK_COLOR: good
+          SLACK_MESSAGE: Nightly E2E (copy) tests passed on master on v0.34.x
```
Comment (erikgrinaker) [inline .github/workflows/e2e-nightly-34x.yml:71]:
```suggestion
          SLACK_MESSAGE: Nightly E2E tests passed on v0.34.x
```

**Review by erikgrinaker — APPROVED** [review]

File: .github/workflows/e2e-nightly-34x.yml (hunk 3 above)
Comment (tessr) [reply .github/workflows/e2e-nightly-34x.yml:71]: Grah, thanks!

File: .github/workflows/e2e-nightly-master.yml (hunk 1 above)
Comment (tessr) [reply .github/workflows/e2e-nightly-master.yml:2]: Ah yes, thanks!

**Pull request merged**